src/vmm/src/guest_config/common/mod.rs

```diff
@@ -0,0 +1,141 @@
+}
+
+impl CPUConfiguration {
+    /// Factory method to instantiate CPUConfiguration
+    /// with the provided `RegisterSet`
+    pub fn new(register_config: HashSet<RegisterSet>) -> CPUConfiguration {
+        CPUConfiguration { register_config }
+    }
+}
+
+/// Template indicating a modification of CPU configuration
+/// to be applied against the `CPUConfiguration` type
+#[allow(unused)]
+pub struct CPUTemplate {
+    /// Modifier masks with pointer data for the modifiers
+    /// intended registers
+    pub modifiers: HashSet<RegisterModifierSet>,
+}
+
+/// Modifier for a CPU configuration containing
+/// the location of the value to be modified,
+/// and provides a value to be applied alongside
+/// a filter to be applied
+#[allow(unused)]
+pub struct RegisterModifier<X, T: Numeric> {
+    /// Location of the register to be modified
+    /// by the mask value and filter
+    pub pointer: X,
+    /// Mask value to be applied
+    pub value: T,
+    /// Mask value is to be applied according
+    /// to what is allowed through by the filter
+    pub filter: T,
+}
+
+/// Numeric wrapper type acting as a marker type
+/// for the register's pointer
+#[allow(unused)]
+pub struct RegisterPointer<T: Numeric> {
+    /// Pointer(location) of a specific register
+    pub pointer: T,
+}
+
+/// Numeric wrapper type acting as a marker type
+/// for the register's value
+#[allow(unused)]
+pub struct RegisterValue<T: Numeric> {
+    /// Value of a specific register
+    pub value: T,
+}
+
+/// CPUID register enumeration
+#[allow(unused)]
+#[allow(missing_docs)]
+#[cfg(target_arch = "x86_64")]
+pub enum CpuidRegister {
+    EAX,
+    EBX,
+    ECX,
+    EDX,
+}
+
+/// Composite type that holistically provides
+/// the location of a specific register being used
+/// in the context of a CPUID tree
+#[allow(unused)]
+#[cfg(target_arch = "x86_64")]
+pub struct CpuidRegisterPointer {
+    leaf: RegisterPointer<u32>,
+    subleaf: RegisterPointer<u32>,
+    pointer: CpuidRegister,
+}
+
+/// Enumerates the types of registers and contains
+/// a property capable of containing the data
+/// formatted for the register type
+#[allow(unused)]
+#[allow(missing_docs)]
+pub enum RegisterSet {
+    #[cfg(target_arch = "x86_64")]
+    CpuId(Cpuid),
+    #[cfg(target_arch = "x86_64")]
+    Msrs(MsrRegisterMap),
+    #[cfg(target_arch = "aarch64")]
+    Aarch64Registers(Aarch64RegisterMap),
+}
+
+/// Intended to correlate to the RegisterSet.
+/// Enumerates the types of register modifiers
+/// with data intended to modify the correlated
+/// registers
+#[allow(unused)]
+#[allow(missing_docs)]
+pub enum RegisterModifierSet {
+    #[cfg(target_arch = "x86_64")]
+    CpuIdModifierSet(HashSet<RegisterModifier<CpuidRegisterPointer, u32>>),
+    #[cfg(target_arch = "x86_64")]
+    MsrsModifierSet(HashSet<RegisterModifier<RegisterPointer<u32>, u64>>),
+    #[cfg(target_arch = "aarch64")]
+    Aarch64RegistersModifierSet(HashSet<RegisterModifier<RegisterPointer<u32>, u64>>),
```

Comment:
should it be "u128>>"?
Would it be too much if we have named types for all pointers and values, eg
```
type MsrId = u32;
type MsrVal = u64;
```

Reply:
It should be 128. I have updated that. With regards to the named types, I'm not sure they add enough value to warrant it.